src/dex/swaap-v2/swaap-v2.ts

```diff
@@ -0,0 +1,875 @@
+        normalizedDestToken.address,
+      );
+
+      if (normalizedSrcToken.address === normalizedDestToken.address) {
+        return null;
+      }
+
+      const pools =
+        limitPools ??
+        (await this.getPoolIdentifiers(srcToken, destToken, side, blockNumber));
+
+      const levels = await this.getCachedLevels();
+      if (levels === null) {
+        return null;
+      }
+
+      const poolIdentifier = getPoolIdentifier(
+        this.dexKey,
+        normalizedSrcToken.address,
+        normalizedDestToken.address,
+      );
+
+      const levelEntries: SwaapV2PriceLevels[] = Object.keys(levels)
+        .map((pair: string) => {
+          if (pools.includes(poolIdentifier)) {
+            return levels[pair];
+          }
+          return undefined;
+        })
+        .filter((o): o is SwaapV2PriceLevels => o !== undefined);
+
+      const unitVolume = getBigNumberPow(
+        (side === SwapSide.SELL ? normalizedSrcToken : normalizedDestToken)
+          .decimals,
+      );
+
+      const amountsFloat = amounts.map(a =>
+        new BigNumber(a.toString()).dividedBy(unitVolume),
+      );
+
+      const prices = levelEntries.map((askAndBids: SwaapV2PriceLevels) => {
+
+        const unitPrice = this.computePricesFromLevelsBids(
+          [unitVolume],
+          askAndBids,
+          normalizedSrcToken,
+          normalizedDestToken,
+          side,
+        )[0];
+        const prices = this.computePricesFromLevelsBids(
+          amountsFloat,
+          askAndBids,
+          normalizedSrcToken,
+          normalizedDestToken,
+          side,
+        );
+
+        return {
+          gasCost: GAS_COST_ESTIMATION,
+          exchange: this.dexKey,
+          data: {},
+          prices,
+          unit: unitPrice,
+          poolIdentifier: requestedPoolIdentifier,
+        } as PoolPrices<SwaapV2Data>;
+      });
+
+      return prices;
+    } catch (e) {
+      this.logger.error(
+        `Error_getPrices ${srcToken}, ${destToken}, ${side}:`, e,
+      );
+      return null;
+    }
+  }
+
+  getBaseToken(poolIdentifier: string): string {
+    return poolIdentifier.split('_')[1];
+  }
+
+  getLevels(
+    askAndBids: SwaapV2PriceLevels,
+    side: SwapSide,
+    poolIdentifier: string,
+  ) {
+    const baseToken: string = this.getBaseToken(poolIdentifier);
+    if (side === SwapSide.SELL) {
+      return askAndBids.asks;
+    }
+    return askAndBids.asks;
+  }
+
+  async preProcessTransaction(
+    optimalSwapExchange: OptimalSwapExchange<SwaapV2Data>,
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    options: PreprocessTransactionOptions,
+  ): Promise<[OptimalSwapExchange<SwaapV2Data>, ExchangeTxInfo]> {
+
+    if(await this.isRestricted()) {
+      this.logger.warn(
+        `${this.dexKey}-${this.network}: protocol is restricted.`,
+      );
+      throw new Error(
+        `${this.dexKey}-${
+          this.network
+        }: protocol is restricted`,
+      );
+    }
+
```

Comment:
I think this is unnecessary and better to remove, because restriction is too prevent failing transaction, but if we are already building transaction, we should try best effort for building transaction

Reply:
@Verisana updated